improver_tests/developer_tools/__init__.py

```diff
@@ -0,0 +1,228 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Set up fixtures for metadata interpreter tests"""
+
+from datetime import datetime
+
+import iris
+import numpy as np
+import pytest
+
+from improver.developer_tools.metadata_interpreter import MOMetadataInterpreter
+from improver.spotdata.build_spotdata_cube import build_spotdata_cube
+from improver.synthetic_data.set_up_test_cubes import (
+    construct_scalar_time_coords,
+    set_up_percentile_cube,
+    set_up_probability_cube,
+    set_up_variable_cube,
+)
+from improver.wxcode.utilities import weather_code_attributes
+
+
+def _update_blended_time_coords(cube):
+    """Replace forecast period and forecast reference time with blend time
+    on blended cubes (to be updated in setup utilities once standard is
+    fully implemented).  Modifies cube in place."""
+    blend_time = cube.coord("forecast_reference_time").copy()
+    blend_time.rename("blend_time")
+    cube.add_aux_coord(blend_time)
+    cube.remove_coord("forecast_period")
+    cube.remove_coord("forecast_reference_time")
+
+
+@pytest.fixture(name="ensemble_cube")
+def ensemble_fixture():
+    """Raw air temperature ensemble in realization space"""
+    data = 285 * np.ones((3, 3, 3), dtype=np.float32)
+    attributes = {
+        "source": "Met Office Unified Model",
+        "title": "MOGREPS-UK Model Forecast on 2 km Standard Grid",
+        "institution": "Met Office",
+        "mosg__model_configuration": "uk_ens",
+    }
+    return set_up_variable_cube(data, attributes=attributes)
+
+
+@pytest.fixture(name="interpreter")
+def interpreter_fixture():
+    return MOMetadataInterpreter()
+
+
+@pytest.fixture(name="landmask_cube")
```

Comment:
Suggest skipping explicit naming of these fixtures as it's not needed here. The reason for using `name` is where a fixture is defined in the same file where it is used, linter tools will complain about the name shadowing.
For example, this one can become:
```python
@pytest.fixture
def landmask_cube():
```

Reply:
That's useful to know, thanks.  I think I'd like to keep the separate names here, because:

- The additional detail in the cube names helps developers interpret specific tests, but
- Putting the additional detail in human-readable form into the pytest fixture names makes the distinctions and links between related fixtures (eg `probability above` and `probability below`) more difficult to "see" when glancing over that file

So I'd quite like to keep these as they are, with the different names serving different purposes in different places.

Reply:
If the explicit/separate fixture naming is intentional and for readability or searchability purposes, I think that's fine.